ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake

```diff
@@ -88,18 +88,22 @@ function(ament_target_dependencies target)
         list_append_unique(link_flags ${${package_name}_LINK_FLAGS})
       endif()
     endforeach()
-    target_compile_definitions(${target}
-      ${required_keyword} ${definitions})
+    if(NOT ARG_INTERFACE)
```

Comment:
is there package which is a interface only package, to sanity check this logic?

Reply:
E.g. ros/pluginlib#188.